### PR TITLE
feat(test): add compactor to deterministic test

### PR DESCRIPTION
**Pull request by wangrunji0408**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This PR adds compactor node to deterministic e2e test and makes sure we use the same configuration as the normal e2e test.
This PR also disables `moka` cache and `RwTracingService` in simulation test. Because they will internally create system threads that bring potential uncertainties and resource leak thus should not be used in the simulation. 
Besides, we will run deterministic tests multiple times on CI, trying to catch those rarely occurring bugs.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: ci/scripts/deterministic-e2e-test.sh [hunk 1]
```diff
@@ -5,30 +5,35 @@ set -euo pipefail
 
 source ci/scripts/common.env.sh
 
+apt-get -y install parallel
```
Comment (BugenZhao) [inline ci/scripts/deterministic-e2e-test.sh:8]: How about make it pre-installed in the CI image?

**Review by BugenZhao — APPROVED** [review]

File: ci/scripts/deterministic-e2e-test.sh (hunk 1 above)
Comment (skyzh) [reply ci/scripts/deterministic-e2e-test.sh:8]: +1, just move it in Dockerfile

**Review by skyzh — APPROVED** [review]
LGTM except the `apt install` should be moved to Dockerfile

File: ci/scripts/deterministic-e2e-test.sh (hunk 1 above)
Comment (wangrunji0408) [reply ci/scripts/deterministic-e2e-test.sh:8]: Should I do something else like building and publishing a docker image?

**Comment by skyzh** [comment]
After you've updated any software in build env, you'll need to bump image version so as to rebuild everything. See https://github.com/singularity-data/risingwave/pull/4478 for example.

**Comment by skyzh** [comment]
Sorry I found that I also gets things wrong in https://github.com/singularity-data/risingwave/pull/4478. Please refer to https://github.com/singularity-data/risingwave/pull/3611/ instead.

**Pull request merged**